src/components/date/date.component.tsx

```diff
@@ -21,26 +20,88 @@ import {
 } from "./__internal__/utils";
 import useLocale from "../../hooks/__internal__/useLocale";
 import Events from "../../__internal__/utils/helpers/events";
-import {
-  filterStyledSystemMarginProps,
-  filterOutStyledSystemSpacingProps,
-} from "../../style/utils";
+import { filterOutStyledSystemSpacingProps } from "../../style/utils";
 import getFormatData from "./__internal__/date-formats";
 import StyledDateInput from "./date.style";
-import Textbox from "../textbox";
-import DatePicker from "./__internal__/date-picker";
+import Textbox, { TextboxProps } from "../textbox";
+import DatePicker, { PickerProps } from "./__internal__/date-picker";
 import DateRangeContext from "../date-range/date-range.context";
 import useClickAwayListener from "../../hooks/__internal__/useClickAwayListener";
 import Logger from "../../__internal__/utils/logger";
 import useFormSpacing from "../../hooks/__internal__/useFormSpacing";
 
-const marginPropTypes = filterStyledSystemMarginProps(
-  styledSystemPropTypes.space
-);
+interface CustomDateEvent {
+  type: string;
+  target: {
+    id?: string;
+    name?: string;
+    value: string;
+  };
+}
+
+export interface DateChangeEvent {
+  target: {
+    id?: string;
+    name?: string;
+    value: {
+      formattedValue: string;
+      rawValue: string | null;
+    };
+  };
+}
+
+export interface DateInputProps
+  extends Omit<
+    TextboxProps,
+    | "value"
+    | "formattedValue"
+    | "rawValue"
+    | "onChange"
+    | "onBlur"
+    | "onMouseDown"
+    | "onChangeDeferred"
+    | "deferTimeout"
+    | "children"
+    | "leftChildren"
+    | "placeholder"
+    | "iconOnClick"
+    | "iconOnMouseDown"
+    | "enforceCharacterLimit"
+    | "characterLimit"
+    | "warnOverLimit"
+    | "iconTabIndex"
+    | "inputIcon"
+  > {
+  /** Boolean to allow the input to have an empty value */
+  allowEmptyValue?: boolean;
+  /** Boolean to toggle where DatePicker is rendered in relation to the Date Input */
+  disablePortal?: boolean;
+  /** Minimum possible date YYYY-MM-DD */
+  minDate?: string;
+  /** Maximum possible date YYYY-MM-DD */
+  maxDate?: string;
+  /** Specify a callback triggered on change */
+  onChange: (ev: DateChangeEvent) => void;
+  /** Specify a callback triggered on blur */
+  onBlur?: (ev: DateChangeEvent) => void;
+  /** The current date string */
+  value: string;
+  /**
+   * Pass any props that match the DayPickerProps interface to override default behaviors
+   * See [DayPickerProps](https://react-day-picker-v7.netlify.app/docs/getting-started/) for a full list of available props
+   * */
+  pickerProps?: PickerProps;
+  /**
+   * @private
+   * @ignore
+   * Name passed from DateRange to allow it to know which input is updating
+   * */
+  inputName?: "start" | "end";
```

Comment:
```suggestion
  inputName?: InputName;
```

**suggestion**: similarly to the couple of uses in the DateRange spec file (again it will have to be imported)